### DISCUSSION TITLE
Silent installation and configuration of PlaneWave software components

### DISCUSSION THREAD

**Discussion by blumzi** [blumzi/MAST_unit#5] [Ideas]
The **MAST** project involves 20 individual `unit` computers, one per telescope.

We need to consider:
* Initial provisioning - installation  and configuration of software
* Maintenance - upgrading of software components, changes of configuration

We prefere to do all these via AUTOMATION tools (e.g. _Windows MDM_ and _Windows Configuration Designer_).

With this in mind we need:
* Silent installers for PlaneWave software (simulators, ASCOM drivers, PWI4, etc.)
* Locations, format and contents of all the PlaneWave components` configuration files (e.g. what ASCOM camera driver is used by autofocus)
* Silent (no GUI) versions of PWI4, plate solver, etc.

**Comment by kivarsen** [comment]
I use the Inno installer for all of our software. The installers take the /silent command line parameter which (assuming the installer is run with Admin privileges to avoid the UAC prompt) will perform the installation without user intervention. The /verysilent parameter allows it to run without even showing a progress window.

The full list of accepted parameters can be found here:
https://jrsoftware.org/ishelp/index.php?topic=setupcmdline

Some of our software, like the PlateSolve tool, don't need to register anything with the system (like ASCOM drivers) and therefore probably do not need actual installers. In these cases I typically just distribute a zip archive with the program and associated files, and the user can place the contents wherever it makes sense on that system.

The config files used by PWI4 can all be found under:

Documents\PlaneWave Instruments\PWI4\Settings\

The config file format is a simple list of [Name] = [Value] pairs. I don't currently have documentation for each entry, but I'm hoping the names and values are self-explanatory enough to be useful. The file PWI4.cfg is always loaded, and some hardware selection entries result in hardware-specific settings being loaded from a separate config file. For example, the PWI4.cfg entry:

Devices.Camera1                     = ASCOM.Camera_1

will cause the config file ASCOM.Camera_1.cfg to be loaded with the ASCOM-specific settings (mainly the COM ID of the ASCOM driver to be used).

I'd be interested in hearing feedback on whether it will be sufficient to pre-generate a set of template config files from the GUI, and then inject new values programmatically on a node-by-node basis. If there are specific entries that need further description or documentation I'd be happy to provide them, but I'm wondering if it might be overkill to document every single entry.
